### DISCUSSION TITLE
Add part of baselib functions only

### DISCUSSION THREAD

**Discussion by wyxloading** [yuin/gopher-lua#393] [Ideas]
- [x] GopherLua is a Lua5.1 implementation. You should be familiar with Lua programming language. Have you read [Lua 5.1 reference manual](http://www.lua.org/manual/5.1/) carefully?
- [x] GopherLua is a Lua5.1 implementation.  In Lua, to keep it simple, **it is more important to remove functionalities rather than to add functionalities unlike other languages** . If you are going to introduce some new cool functionalities into the GopherLua code base and the functionalities can be implemented by existing APIs, It should be implemented as a library. 

Please answer the following before submitting your issue:

1. What version of GopherLua are you using? : 1cd887c
2. What version of Go are you using? : 1.13
3. What operating system and processor architecture are you using? : linux/amd64
4. What did you do? : N/A
5. What did you expect to see? : N/A
6. What did you see instead? : N/A

I want to add part of the baselib functions to LState, but there is no way to do that since every baselib function implementation is private.  
What about add something like `OpenBaseStrip` which can pass a list of baselib function name, and add them to LState.

**Comment by tul** [comment]
As a workaround, after calling `OpenBase` you can then access the table in your lua state and just nil out the unwanted functions before loading your lua code into it.

**Comment by ShadowNineX** [comment]
I just did this: 
```go
func safeLuaState() *lua.LState {
	L := lua.NewState(lua.Options{
		SkipOpenLibs: true, // skip loading all libs by default
	})

	// Open safe standard libraries explicitly
	lua.OpenBase(L)   // basic functions like print, type, error
	lua.OpenTable(L)  // table library
	lua.OpenString(L) // string library
	lua.OpenMath(L)   // math library

	// Note: io and os are NOT opened -> sandboxing

	// Override Lua's print to prefix output
	L.SetGlobal("print", L.NewFunction(func(L *lua.LState) int {
		top := L.GetTop()
		args := make([]any, 0, top)
		for i := 1; i <= top; i++ {
			args = append(args, L.ToStringMeta(L.Get(i)).String())
		}
		allArgs := make([]any, 0, len(args)+1)
		allArgs = append(allArgs, "[LUA]")
		allArgs = append(allArgs, args...)
		log.Println(allArgs...)
		return 0
	}))

	return L
}

func main() {
	L := safeLuaState()
	defer L.Close()

	script := `
		print("Hello from Lua!")
		if io == nil then
			print("io library is blocked!")
		end
		if os == nil then
			print("os library is blocked!")
		end
	`

	if err := L.DoString(script); err != nil {
		log.Fatalf("Lua error: %v", err)
	}
}
```
